build/testing/sauce-labs/test-browsers.js

```diff
@@ -0,0 +1,196 @@
+                        await driver.navigate().to(domain + "/components/metatext/");
+                        await driver.navigate().to(domain + "/components/paragraph/");
+                        await driver.navigate().to(domain + "/components/subheading/");
+                        await driver.navigate().to(domain + "/components/text-field/");
+                        await driver.navigate().to(domain + "/components/toggle/");
+                        await driver.navigate().to(domain + "/components/typography/");
+
+                        // Turn on Developer tools and iterate each component
+                        // xpath selectors do not work across browsers. This in particular breaks in Internet Explorer
+                        // await driver.findElement(By.xpath("//button[text()='dev tools']")).click();
+                        // one possible solution would be to inject data attributes for each component we can bind to.
+                        // For example, data-test="friendly name"
+                        // driver.findElement(By.cssSelector('[data-element="city"]'))
+
+                        // Navigate to example views for each component
+                        await driver.navigate().to(domain + "/components/button/examples");
+                        await driver.navigate().to(domain + "/components/caption/examples");
+                        await driver.navigate().to(domain + "/components/checkbox/examples");
+                        await driver.navigate().to(domain + "/components/dialog/examples");
+                        await driver.navigate().to(domain + "/components/divider/examples");
+                        await driver.navigate().to(domain + "/components/flipper/examples");
+                        await driver.navigate().to(domain + "/components/heading/examples");
+                        await driver.navigate().to(domain + "/components/hypertext/examples");
+                        await driver.navigate().to(domain + "/components/image/examples");
+                        await driver.navigate().to(domain + "/components/label/examples");
+                        await driver.navigate().to(domain + "/components/metatext/examples");
+                        await driver.navigate().to(domain + "/components/paragraph/examples");
+                        await driver.navigate().to(domain + "/components/subheading/examples");
+                        await driver.navigate().to(domain + "/components/text-field/examples");
+                        await driver.navigate().to(domain + "/components/toggle/examples");
+                        await driver.navigate().to(domain + "/components/typography/examples");
+
+                } finally {
+                        await driver.quit();
+                }
+        })
+    });
+}
+
+/**
+ * Get Branch Name using child_process and Git which runs Asynchronously
+ */
+var getBranchName = new Promise(function(resolve, reject) {

```

Comment:
this block reads a bit weird to me. Perhaps instead of getBranchName, we rename this to be initializeTest? Then we can chain the "then" directly with the promise and call initializeTest?

```
var initializeTest = new Promise(function(resolve, reject) {
    git.stdout.on('data', (data) => {
        resolve(`${data}`);
    });
}).then(function(branchName) {
     // slice off '/n' at the end of string
    branchName = branchName.slice(0, -1);
     // Run Testing
    test(branchName);
});

initializeTest();
```

Reply:
@nicholasrice or @janechu for thoughts

Reply:
But, this isn't initializing.  This is literally getting the current branch name.

Reply:
I think if you want to go further you could skip the variable assignment so this should work:

```js
/**
* Execute tests against the branch
*/
new Promise(function(resolve, reject) {
    git.stdout.on('data', (data) => {
        resolve(`${data}`.trim());
    });
}).then(function(branchName) {
    test(branchName);
});
```